lib/happo/views/diffs.erb

```diff
@@ -10,52 +10,27 @@
     <style>
       <%= Happo::Utils.css_styles %>
     </style>
-  </head>
-
-  <body>
-    <header class="header">
-      <h1 class="header_title">
-        <%= Happo::Utils.page_title(diff_images, new_images) %>
-      </h1>
-      <div class="header__time">Generated: <%= Time.now %></div>
-    </header>
-
-    <main class="main">
-      <% if diff_images.count > 0%>
-        <h2 id="diffs">
-          <a class="anchored" href="#diffs">
-            Diffs (<%= diff_images.count %>)
-          </a>
-        </h2>
 
-        <% diff_images.each do |image| %>
-          <h3 id="<%= Happo::Utils.image_slug(image) %>">
-            <a class="anchored" href="#<%= Happo::Utils.image_slug(image) %>">
-              <%= Rack::Utils.escape_html(image[:description]) %> @ <%= image[:viewport] %>
-            </a>
-          </h3>
+    <script src="https://npmcdn.com/react@15.3.1/dist/react.min.js"></script>
+    <script src="https://npmcdn.com/react-dom@15.3.1/dist/react-dom.min.js"></script>
+    <script src="https://npmcdn.com/babel-core@5.8.38/browser.min.js"></script>
 
-          <img src="<%= image[:url] %>">
-        <% end %>
-      <% end %>
-
-      <% if new_images.count > 0%>
-        <h2 id="new">
-          <a class="anchored" href="#new">
-            New examples (<%= new_images.count %>)
-          </a>
-        </h2>
-
-        <% new_images.each do |image| %>
-          <h3 id="<%= Happo::Utils.image_slug(image) %>">
-            <a class="anchored" href="#<%= Happo::Utils.image_slug(image) %>">
-              <%= Rack::Utils.escape_html(image[:description]) %> @ <%= image[:viewport] %>
-            </a>
-          </h3>
+    <script type="text/babel">
+      <%= Happo::Utils.jsx_code %>
+    </script>
+  </head>
 
-          <img src="<%= image[:url] %>">
-        <% end %>
-      <% end %>
-    </main>
+  <body>
+    <script type="text/babel">
+      ReactDOM.render(
+        <HappoDiffs
+          newImages={<%= new_images.to_json %>}
+          diffImages={<%= diff_images.to_json %>}
+          pageTitle={<%= Happo::Utils.page_title(diff_images, new_images).to_json %>}
+          generatedAt={<%= Time.now.to_s.to_json %>}
+        />,
+        document.body
```

Comment:
I've read that people suggest not rendering directly into the body because it doesn't work very well with some things. It might be best to render into a div.